src/Microsoft.TemplateEngine.Core/Conditional.cs

```diff
@@ -2,95 +2,191 @@
 using System.IO;
 using System.Text;
 using Microsoft.TemplateEngine.Abstractions.Engine;
+using System;
+using System.Runtime.CompilerServices;
 
 namespace Microsoft.TemplateEngine.Core
 {
     public class Conditional : IOperationProvider
     {
-        private readonly string _elseIfToken;
-        private readonly string _elseToken;
-        private readonly string _endIfToken;
-        private readonly string _ifToken;
         private readonly ConditionEvaluator _evaluator;
         private readonly bool _wholeLine;
         private readonly bool _trimWhitespace;
 
-        public string IfToken => _ifToken;
+        private readonly IList<string> _ifTokens = new List<string>();
```

Comment:
Makes sense, good idea. I’ve removed all the intermediate storage.

From: Mike Lorbetske [mailto:notifications@github.com]
Sent: Tuesday, August 23, 2016 11:09 AM
To: seancpeters/microsoft-templateengine <microsoft-templateengine@noreply.github.com>
Cc: Sean Peters <sepeters@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [seancpeters/microsoft-templateengine] Refactor (#1)


In src/Microsoft.TemplateEngine.Core/Conditional.cs<https://github.com/seancpeters/microsoft-templateengine/pull/1#discussion_r75920637>:

>          private readonly ConditionEvaluator _evaluator;

>          private readonly bool _wholeLine;

>          private readonly bool _trimWhitespace;

>

> -        public string IfToken => _ifToken;

> +        private readonly IList<string> _ifTokens = new List<string>();

We should just be able to hang on to the config that wraps up all the params

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/seancpeters/microsoft-templateengine/pull/1/files/a425b2f9e55c8aa446463bd90060cb6bc99db3d0#r75920637>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AT8a8WPoprDtJcphOx8Rc2iBd809gZLQks5qizdNgaJpZM4JqaeQ>.